pyquil/quil.py

```diff
@@ -100,6 +102,27 @@
 ]
 
 
+RetType = TypeVar("RetType")
+
+
+def _invalidates_cached_properties(func: Callable[..., RetType]) -> Callable[..., RetType]:
```

Comment:
Is the performance boost you got worth the risk of this?

Seems like there should be a developer note somewhere to prevent this decorator being left off of somewhere it's needed as the class evolves

Reply:
Yeah, the performance boost is significant, but probably for too narrow of a use case. Maybe this is better to keep in the back pocket, then we can pull it out if it's really needed.